### PR TITLE
Print stack trace before the server crashes

### PR DESCRIPTION
**Pull request by krishung5**

### CODE REVIEW

File: src/servers/signal.cc [hunk 1]
```diff
@@ -107,6 +110,8 @@ void
 SignalHandler(int signum)
 {
   std::cout << "Signal (" << signum << ") received." << std::endl;
+  std::cout << boost::stacktrace::stacktrace() << std::endl;
```
Comment (deadeyegoodwin) [inline src/servers/signal.cc:113]: Don't dump stack trace for the existing signals that are handled...

File: src/servers/signal.cc [hunk 2]
```diff
@@ -115,9 +120,12 @@ SignalHandler(int signum)
 TRITONSERVER_Error*
 RegisterSignalHandler()
 {
-  // Trap SIGINT and SIGTERM to allow server to exit gracefully
+  // Trap SIGINT, SIGTERM, SIGSEGV and SIGABRT to allow server to exit
+  // gracefully
   signal(SIGINT, SignalHandler);
   signal(SIGTERM, SignalHandler);
+  signal(SIGSEGV, SignalHandler);
```
Comment (deadeyegoodwin) [inline src/servers/signal.cc:127]: For SEGV and ABRT we don't want to use the existing signalhandler that tries to exit gracefully. Instead we should use a new handler for those that just dumps the stack trace and then does exit(1)

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/servers/signal.cc (hunk 2 above)
Comment (krishung5) [reply src/servers/signal.cc:127]: Got it, will modify.

File: src/servers/signal.cc [hunk 3]
```diff
@@ -110,6 +113,15 @@ SignalHandler(int signum)
   CommonSignalHandler();
 }
 
+void
+ErrorSignalHandler(int signum)
+{
+  std::cout << "Signal (" << signum << ") received." << std::endl;
```
Comment (deadeyegoodwin) [inline src/servers/signal.cc:119]: Missed this before, output to cerr

**Review by deadeyegoodwin — CHANGES_REQUESTED** [review]

File: src/servers/signal.cc (hunk 3 above)
Comment (krishung5) [reply src/servers/signal.cc]: Updated.

**Review by deadeyegoodwin — APPROVED** [review]
Make sure to run CI for this

**Comment by krishung5** [comment]
@deadeyegoodwin I have a CI running now, will not merge it in until passes.

**Pull request merged**